### DISCUSSION TITLE
Задание 3

### DISCUSSION THREAD

**Discussion by eprilutskaya** [hamenok/amasty#8] [General]
**Система управления настройками модуля**

https://github.com/hamenok/amasty/blob/0f2620059aac6ecd61efcaa4d21e8d694c7922e7/Hamenok/Controller/Index/Index.php#L32
Если название самого метода конфиг провайдера подразумевает получение значения определенного конфига, то правильнее в нем при вызове isSetFlag передавать path, а не передавать параметр path в метод, который вызывается public у конфиг провайдера, пример:
```
    public function isEnabled(): bool
    {
        return $this->isSetFlag("general/enabled");
    }

    public function getWelcomeMessage(): string
    {
        return $this->getValue("general/welcome_text");
    }
```
Сам path так же стоит выносить в константы const конфиг провайдера. Текст, если в нем не используются переменные, правильнее ограничивать в одинарные кавычки 'blabla'.

https://github.com/hamenok/amasty/blob/0f2620059aac6ecd61efcaa4d21e8d694c7922e7/Hamenok/Block/Form/Form.php#L30
Если метод возвращает значение, то его название правильно начинать с get getDefaultValueQty

В темплите можно использовать конструкцию 
```
<?php if (условие): ?>
// html который нужно вывести, может содержать так же php вставки 
<?php endif; ?>
```

**Разбор основных компонентов Magento**

https://github.com/hamenok/amasty/blob/0f2620059aac6ecd61efcaa4d21e8d694c7922e7/Hamenok/Controller/Cart/Add.php#L55
Перед получением продукта стоит проверить значения параметров $skuProduct и $skuProduct , если они пустые, то выполнить редирект с ошибкой, что не все данные указали.

https://github.com/hamenok/amasty/blob/0f2620059aac6ecd61efcaa4d21e8d694c7922e7/Hamenok/Controller/Cart/Add.php#L63
Получение квоты правильнее делать уже в if, тк до этого выполняются проверки продукта и до добавления в корзину может не дойти, получится что по коду было выполнено лишнее действие.
if (!$quote->getId()) {

Оценка 9

**Comment by hamenok** [comment]
исправил :)
